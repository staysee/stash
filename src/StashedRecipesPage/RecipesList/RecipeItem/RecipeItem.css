.RecipeItem {
    border: 1px solid gray;
    cursor: pointer;
    display: flex;
    flex-direction: column;
    margin-bottom: 15px;
    padding: 5px;
    width: 45%;
}

.RecipeItem p {
    margin: 0;
    padding: 10px;
    text-transform: uppercase;
}