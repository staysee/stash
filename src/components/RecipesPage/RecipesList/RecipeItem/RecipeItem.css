.RecipeItem {
  border: 1px solid gray;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
  cursor: pointer;
  display: flex;
  flex-direction: column;
  margin-bottom: 15px;
  padding: 5px;
  width: 45%;
}

.RecipeItem:hover {
  box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
  cursor: pointer;
}

.RecipeItem .item_image {
  background-color: #ebebeb;
  border: 2px solid #ffffff;
  display: flex;
  height: 200px;
}

.RecipeItem img {
  max-height: 100%;
  max-width: 100%;
  margin: 0 auto;
}

.RecipeItem p {
  margin: 0;
  padding: 10px;
  text-transform: uppercase;
}
