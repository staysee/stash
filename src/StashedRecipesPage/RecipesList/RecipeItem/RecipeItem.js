import React from 'react'

import './RecipeItem.css'

function RecipeItem({id, title, ingredients, instructions, imageURL, handleOpenModal}){
    return (
        <>
            <li 
                className="RecipeItem" 
                key={id}
                onClick={handleOpenModal}
            >
                <img src={imageURL} alt={title} />
                <p>{title}</p>
            </li>
        </>   
    )
}

export default RecipeItem