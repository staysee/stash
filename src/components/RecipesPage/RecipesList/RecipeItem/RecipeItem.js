import React from 'react';
import PropTypes from 'prop-types';
import './RecipeItem.css';

function RecipeItem({ id, title, ingredients, instructions, imageURL, handleOpenModal }) {
  return (
    <>
      <li className="RecipeItem" key={id} onClick={handleOpenModal}>
        <div className="item_image">
          <img src={imageURL} alt={title} />
        </div>
        <p>{title}</p>
      </li>
    </>
  );
}

export default RecipeItem;

RecipeItem.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  ingredients: PropTypes.string,
  instructions: PropTypes.string,
  imageUrl: PropTypes.string,
  handleOpenModal: PropTypes.func,
};
