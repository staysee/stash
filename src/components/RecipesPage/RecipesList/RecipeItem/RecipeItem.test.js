import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import RecipeItem from './RecipeItem';

describe (`Recipe Item component`, () => {
    const props = {
        id: 1,
        title: 'Title',
        ingredients: 'List of ingredients',
        instructions: 'Set of instructions',
        imageUrl: 'https://bit.ly/3cotna1',
    }

    it ('renders without crashing', () => {
        // create DOM element to render the component into
        const div = document.createElement('div');
        // render the component, this is the actual test, if something is wrong it will show up here
        ReactDOM.render(<RecipeItem />, div);  
        // clean up code
        ReactDOM.unmountComponentAtNode(div);
    })

    it (`renders the UI as expected`, () => {
        const tree = renderer
            .create(<RecipeItem {...props} />)
            .toJSON();
        expect(tree).toMatchSnapshot()
    })

})