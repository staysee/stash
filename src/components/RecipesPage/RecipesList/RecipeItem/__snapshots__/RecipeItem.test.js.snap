// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Item component renders the UI as expected 1`] = `
<li
  className="RecipeItem"
>
  <div
    className="item_image"
  >
    <img
      alt="Title"
    />
  </div>
  <p>
    Title
  </p>
</li>
`;
